crates/bevy_transform/src/lib.rs

```diff
@@ -4,16 +4,129 @@ pub mod transform_propagate_system;
     #[doc(hidden)]
-    pub use crate::{components::*, hierarchy::*, TransformPlugin};
+    pub use crate::{components::*, hierarchy::*, TransformBundle, TransformPlugin};
 }
 
 use bevy_app::prelude::*;
+// Note: Neccesary because `Component` is only used in the documentation of TransformBundle.
+#[allow(unused_imports)]
 use bevy_ecs::{
+    bundle::Bundle,
+    component::Component,
     schedule::{ParallelSystemDescriptorCoercion, SystemLabel},
     system::IntoSystem,
 };
+use bevy_math::{Mat4, Quat, Vec3};
 use prelude::{parent_update_system, Children, GlobalTransform, Parent, PreviousParent, Transform};
 
+/// A [`Bundle`] of the [`Transform`] and [`GlobalTransform`] [`Component`]s, which describe the position of an entity.
+///
+/// * To place or move an entity, you should set its [`Transform`].
+/// * To get the global position of an entity, you should get its [`GlobalTransform`].
+/// * To be displayed, an entity must have both a [`Transform`] and a [`GlobalTransform`].
+///   * Use the [`TransformBundle`] to guaranty this.
+///
+/// ## [`Transform`] and [`GlobalTransform`]
+///
+/// [`Transform`] is the position of an entity relative to its parent position, or the reference
+/// frame if it doesn't have a [`Parent`](super::Parent).
+///
+/// [`GlobalTransform`] is the position of an entity relative to the reference frame.
+///
+/// [`GlobalTransform`] is updated from [`Transform`] in the system
+/// [`transform_propagate_system`](crate::transform_propagate_system::transform_propagate_system).
+///
+/// In pseudo code:
+/// ```ignore
+/// for entity in entities_without_parent:
+///     set entity.global_transform to entity.transform
+///     recursively:
+///         set parent to current entity
+///         for child in parent.children:
+///             set child.global_transform to parent.global_transform * child.transform
+/// ```
+///
+/// This system runs in stage [`CoreStage::PostUpdate`](crate::CoreStage::PostUpdate). If you
+/// update the[`Transform`] of an entity in this stage or after, you will notice a 1 frame lag
+/// before the [`GlobalTransform`] is updated.
+#[derive(Default, Bundle, Clone, Debug)]
+pub struct TransformBundle {
+    pub local: Transform,
+    pub global: GlobalTransform,
+}
+
+impl TransformBundle {
+    /// Creates a new [`TransformBundle`] at the position `(x, y, z)`. In 2d, the `z` component
+    /// is used for z-ordering elements: higher `z`-value will be in front of lower
+    /// `z`-value.
+    #[inline]
+    pub fn from_xyz(x: f32, y: f32, z: f32) -> Self {
+        TransformBundle {
+            local: Transform::from_xyz(x, y, z),
+            ..Default::default()
+        }
+    }
+
+    /// Creates a new identity [`TransformBundle`], with no translation, rotation, and a scale of 1
+    /// on all axes.
+    #[inline]
+    pub const fn identity() -> Self {
+        TransformBundle {
+            local: Transform::identity(),
+            global: GlobalTransform::identity(),
```

Comment:
```suggestion
            ..Default::default()
```

Reply:
The `GlobalTransform::identity()` is neccesary here, because this function is const but `..Default::default()` isn't.
I will add a comment.